### PR TITLE
feat: events

### PR DESCRIPTION
**Pull request by n1rjal**
- Add events serializer, urls, views
- Add weak name

### CODE REVIEW

File: event/models.py [hunk 1]
```diff
@@ -8,8 +10,31 @@ def __str__(self) -> str:
         return self.name
 
 
+class EventLocation(models.Model):
+    name = models.CharField(max_length=150, blank=False, null=False)
+    # it is nullable as the location may be not always findable
+    google_maps_location = models.URLField(
```
Comment (Sailesh-Singh) [inline event/models.py:16]: Add url validation for google map
eg: ^(https:\/\/)?(www\.)?(google\.com\/maps|goo\.gl\/maps)\/.*$

either in model or serializer to keep rigid contorl, avoid xss, to enhance seo

**Review by rex9840 — APPROVED** [review]

File: event/models.py [hunk 2]
```diff
@@ -8,8 +10,31 @@ def __str__(self) -> str:
...
+class HotTopic(models.Model):
+    name = models.CharField(max_length=100, primary_key=True)
+
+    def __str__(self):
+        return self.name
+
+
 # Create your models here.
 class Event(models.Model):
+    slug = models.SlugField(
```
Comment (Sailesh-Singh) [inline event/models.py:33]: adding fallback of slug if entered repeatedly will be beneficial
pre_save slug for bulk operation

File: event/models.py [hunk 3]
```diff
@@ -8,8 +10,31 @@ def __str__(self) -> str:
...
+        max_length=350,
+        primary_key=True,
+        editable=False,
+    )
     title = models.CharField(max_length=400, blank=False, null=False)
     start_date = models.DateField(blank=False, null=False)
```
Comment (Sailesh-Singh) [inline event/models.py:39]: date validation that end_date is after start_date
registration_deadline fall before start_date

File: event/serializers.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+class ScheduleSerializer(serializers.ModelSerializer):
+    speakers = SpeakerSerializer(many=True)
+
+    class Meta:
+        model = models.Schedule
+        fields = "__all__"
+
+
+class EventSerializer(serializers.ModelSerializer):
```
Comment (Sailesh-Singh) [inline event/serializers.py:37]: Ensure creation and updating of Event model 

eg:
```python
def update(self, instance, validated_data):
        instance.title = validated_data.get('title', instance.title)
        instance.start_date = validated_data.get('start_date', instance.start_date)
        instance.end_date = validated_data.get('end_date', instance.end_date)
        # similarly aru fields
        instance.save()
        return instance
```

**Review by Sailesh-Singh — CHANGES_REQUESTED** [review]

File: event/models.py (hunk 2 above)
Comment (n1rjal) [reply event/models.py:33]: What do you mean ?

File: event/models.py (hunk 1 above)
Comment (n1rjal) [reply event/models.py:16]: I mean admins will put it and google does have shorter URLs as well.

File: event/serializers.py (hunk 4 above)
Comment (n1rjal) [reply event/serializers.py:37]: No my theory is that we will only create events from Admin Panel

File: event/models.py (hunk 1 above)
Comment (Sailesh-Singh) [reply event/models.py:16]: Okay

**Review by Sailesh-Singh — APPROVED** [review]

**Pull request merged**